core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -197,10 +197,17 @@ public static Catalog buildIcebergCatalog(String name, Map<String, String> optio
   /**
    * Load a custom {@link FileIO} implementation.
    * <p>
-   * The implementation must have a no-arg constructor.
+   * The implementation should have a no-arg constructor.
    * If the class implements {@link Configurable},
    * a Hadoop config will be passed using {@link Configurable#setConf(Configuration)}.
-   * {@link FileIO#initialize(Map properties)} is called to complete the initialization.
+   *
+   * For compatibility with HadoopFileIO,
```

Comment:
I wonder if we starts to support 1-arg constructor as a valid case in the code and prints error messages about it, should we mention this as a fully supported case instead of a backward compatibility support? Especially that the reason for you to choose this option is performance concern.

Reply:
Yes I also noticed this is a bit annoying to see a stack trace printed when trying to load it, let me just not print the error.